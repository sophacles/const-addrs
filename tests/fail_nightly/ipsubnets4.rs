use const_addrs::ipsubnets4;

fn main() {
    let _ = ipsubnets4!("10.0.01", "10.0.1.0", 24);
    let _ = ipsubnets4!("10.0.0.1", "10.0.10", 24);
    let _ = ipsubnets4!("10.0.01", "10.0.10", 24);
    let _ = ipsubnets4!("10.0.0.1", "2001:db8::32:23", 24);
    let _ = ipsubnets4!("10.0.0.1", "10.0.10.0", 33);
    let _ = ipsubnets4!("10.0.0.1", "10.0.10.0", 260);
}
