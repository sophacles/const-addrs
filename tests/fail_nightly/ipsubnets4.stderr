error: invalid IP address syntax
 --> tests/fail_nightly/ipsubnets4.rs:4:25
  |
4 |     let _ = ipsubnets4!("10.0.01", "10.0.1.0", 24);
  |                         ^^^^^^^^^

error: invalid IP address syntax
 --> tests/fail_nightly/ipsubnets4.rs:5:37
  |
5 |     let _ = ipsubnets4!("10.0.0.1", "10.0.10", 24);
  |                                     ^^^^^^^^^

error: invalid IP address syntax
 --> tests/fail_nightly/ipsubnets4.rs:6:25
  |
6 |     let _ = ipsubnets4!("10.0.01", "10.0.10", 24);
  |                         ^^^^^^^^^

error: invalid IP address syntax
 --> tests/fail_nightly/ipsubnets4.rs:6:36
  |
6 |     let _ = ipsubnets4!("10.0.01", "10.0.10", 24);
  |                                    ^^^^^^^^^

error: Addr is IPv6 but IPv4 is expected
 --> tests/fail_nightly/ipsubnets4.rs:7:37
  |
7 |     let _ = ipsubnets4!("10.0.0.1", "2001:db8::32:23", 24);
  |                                     ^^^^^^^^^^^^^^^^^

error: Minimum prefix must be 32 or less
 --> tests/fail_nightly/ipsubnets4.rs:8:50
  |
8 |     let _ = ipsubnets4!("10.0.0.1", "10.0.10.0", 33);
  |                                                  ^^

error: number too large to fit in target type
 --> tests/fail_nightly/ipsubnets4.rs:9:50
  |
9 |     let _ = ipsubnets4!("10.0.0.1", "10.0.10.0", 260);
  |                                                  ^^^
